lib/std/time.zig

```diff
@@ -242,15 +242,24 @@ pub const Timer = struct {
 
     fn nativeDurationToNanos(self: Timer, duration: u64) u64 {
         if (is_windows) {
-            return @divFloor(duration * ns_per_s, self.frequency);
+            return safeMulDiv(duration, ns_per_s, self.frequency);
         }
         if (comptime std.Target.current.isDarwin()) {
-            return @divFloor(duration * self.frequency.numer, self.frequency.denom);
+            return safeMulDiv(duration, self.frequency.numer, self.frequency.denom);
         }
         return duration;
     }
 };
 
+// Calculate (a * b) / c without risk of overflowing too early because of the
+// multiplication.
+fn safeMulDiv(a: u64, b: u64, c: u64) u64 {
```

Comment:
```zig
fn safeMulDiv(comptime T: type, a: T, b: T, c: T) T
```
and move to math?

Reply:
It's not meant to be used for anything but this calculation, the safe part in the name refers to the typical values used here.